File: HRMS/HRMS.Web/Modules/ServerTypes/Operations/EmployeeLeaveProfileColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { EmployeeLeaveProfileRow } from "./EmployeeLeaveProfileRow";

export interface EmployeeLeaveProfileColumns {
    EmployeeLeaveProfileId: Column<EmployeeLeaveProfileRow>;
    EmployeeFullName: Column<EmployeeLeaveProfileRow>;
    DepartmentName: Column<EmployeeLeaveProfileRow>;
    DesignationName: Column<EmployeeLeaveProfileRow>;
    LeavePolicyName: Column<EmployeeLeaveProfileRow>;
    LeaveTypeName: Column<EmployeeLeaveProfileRow>;
    OpeningBalance: Column<EmployeeLeaveProfileRow>;
    AccruedLeave: Column<EmployeeLeaveProfileRow>;
    UsedLeave: Column<EmployeeLeaveProfileRow>;
    PendingLeave: Column<EmployeeLeaveProfileRow>;
    CarryForwardLeave: Column<EmployeeLeaveProfileRow>;
    LOPDays: Column<EmployeeLeaveProfileRow>;
    LastUpdatedDate: Column<EmployeeLeaveProfileRow>;
}

export class EmployeeLeaveProfileColumns extends ColumnsBase<EmployeeLeaveProfileRow> {
    static readonly columnsKey = 'Operations.EmployeeLeaveProfile';
    static readonly Fields = fieldsProxy<EmployeeLeaveProfileColumns>();
}

File: HRMS/HRMS.Web/Modules/Operations/EmployeeLeaveProfile/EmployeeLeaveProfileColumns.cs
using Serenity.ComponentModel;
using System.ComponentModel;

namespace HRMS.Operations.Columns;

[ColumnsScript("Operations.EmployeeLeaveProfile")]
[BasedOnRow(typeof(EmployeeLeaveProfileRow), CheckNames = true)]
public class EmployeeLeaveProfileColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int EmployeeLeaveProfileId { get; set; }
    [EditLink]
    public string EmployeeFullName { get; set; }
    public string DepartmentName { get; set; }
    public string DesignationName { get; set; }
    public string LeavePolicyName { get; set; }
    public string LeaveTypeName { get; set; }
    public decimal OpeningBalance { get; set; }
    public decimal AccruedLeave { get; set; }
    public decimal UsedLeave { get; set; }
    public decimal PendingLeave { get; set; }
    public decimal CarryForwardLeave { get; set; }
    public decimal LOPDays { get; set; }
    public DateTime LastUpdatedDate { get; set; }
}
